src/main/java/org/storage/metadata/controller/UserController.java

```diff
@@ -0,0 +1,96 @@
+package org.storage.metadata.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import org.storage.metadata.model.User;
+import org.storage.metadata.repository.UserRepository;
+import org.storage.metadata.worker.dto.LoginDTO;
+import org.storage.metadata.worker.dto.SignUpDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/auth")
+public class UserController {
```

Comment:
We should create two controllers
1. UserLoginController: it can be the current controller, which takes care of sign-in, sign-up, username availability and forgot password
2. UserController: Get user metadata, update user metadata